src/LibraryManager/Manifest.cs

```diff
@@ -462,10 +453,12 @@ public void AddVersion(string version)
         /// <returns></returns>
         public async Task<bool> RemoveUnwantedFilesAsync(Manifest newManifest, CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
+
             if (newManifest != null)
             {
-                ISet<FileIdentifier> existingFiles = await GetAllManifestFilesWithVersionsAsync(Libraries).ConfigureAwait(false);
-                ISet<FileIdentifier> newFiles = await GetAllManifestFilesWithVersionsAsync(newManifest.Libraries).ConfigureAwait(false);
+                IEnumerable<FileIdentifier> existingFiles = await GetAllManifestFilesWithVersionsAsync(Libraries).ConfigureAwait(false);
+                IEnumerable<FileIdentifier> newFiles = await GetAllManifestFilesWithVersionsAsync(newManifest.Libraries).ConfigureAwait(false);
                 IEnumerable<string> filesToRemove = existingFiles.Where(f => !newFiles.Contains(f)).Select(f => f.Path);
```

Comment:
Probably not an issue, just wanted to make sure again that these enumerations aren't going to be large.

Reply:
Yes, these might get big. I will update.

Reply:
Replaced with existingFiles.Except(newFiles).Select(f => f.Path);